pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -271,6 +271,42 @@ impl<T: Config> Pallet<T> {
 		fee
 	}
 
+	/// Compute the length portion of a fee by invoking the configured `LengthToFee` impl.
+	/// # Arguments
+	/// * `length` - The length of the transaction.
+	/// * `weight` - The weight of the transaction.
+	///
+	/// # Returns
+	/// `FeeDetails` - The fee details for the transaction.
+	pub fn compute_capacity_fee_details<
+		Extrinsic: sp_runtime::traits::Extrinsic + GetDispatchInfo,
+	>(
+		unchecked_extrinsic: Extrinsic,
+		len: u32,
+	) -> FeeDetails<BalanceOf<T>>
+	where
+		<T as pallet::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+	{
+		let dispatch_info = <Extrinsic as GetDispatchInfo>::get_dispatch_info(&unchecked_extrinsic);
+		let extrinsic_weight = dispatch_info.weight;
+		let weight_fee = Self::weight_to_fee(extrinsic_weight);
+
+		let len_fee = Self::length_to_fee(len);
+		let base_fee = Self::weight_to_fee(CAPACITY_EXTRINSIC_BASE_WEIGHT);
+
+		let adjusted_weight_fee = base_fee.saturating_add(weight_fee).saturating_add(len_fee);
```

Comment:
I think there is no `adjusted_weight_fee` with Capacity fee.

Reply:
so adjusted_weight_fee here as of now is -> `capacity base fee` + `capacity weight fee` + `len fee`, you mean we dont need  to represent it as adjusted weight fee but rather `base_fee = capacity base fee + weight fee + len fee` , while `tip and adjusted_weight_fee =0` ?

Suggestions

Reply:
@wilwade  any suggestion on this, `FeeDetail` type from transaction pallet has three type of fee how it translate to our fee structure that we want to return?